source/slang/slang-ir-link.cpp

```diff
@@ -89,6 +89,25 @@ struct IRSpecContextBase
     }
 };
 
+enum class GlobalValueClass
+{
+    StructKey,
+    Other
+};
+
+// Get the "class" of a global value. If there are more than one value of the same class,
+// only one value in each class will be selected during linking.
+GlobalValueClass getGlobalValueClass(IRInst* value)
+{
+    switch (value->op)
+    {
+    case kIROp_StructKey:
+        return GlobalValueClass::StructKey;
+    default:
+        return GlobalValueClass::Other;
+    }
+}
+
```

Comment:
This seems like a reasonable workaround for now, but the question I continue to have is: why does anything other than the key for a requirement need a mangled name and/or linkage?

It seems like the fundamental problem here stems from trying to emit extra "stuff" from an interface requirement, and then giving that "stuff" a mangled name (that happens to conflict with the key, which was the thing that was original there and using the mangled name).

I don't see any evidence that the `Other` case here is being used for things that ever need linkage.

Reply:
Done. Reverted.